tests/infra/remote.py

```diff
@@ -278,12 +278,23 @@ def wait_for_stdout_line(self, line, timeout):
                 if stdout.channel.recv_exit_status() == 0:
                     return
                 time.sleep(1)
-            raise ValueError(
-                "{} not found in stdout after {} seconds".format(line, timeout)
-            )
+            raise ValueError(f"{line} not found in stdout after {timeout} seconds")
         finally:
             client.close()
 
+    def check_for_stdout_line(self, line, timeout):
+        try:
+            client = self._connect_new()
+            for _ in range(timeout):
+                _, stdout, _ = client.exec_command(f"grep -F '{line}' {self.root}/out")
+                if stdout.channel.recv_exit_status() == 0:
+                    return True
+                time.sleep(1)
+            return False
+        except Exception:
```

Comment:
We also want to close the client if there's no error!

Reply:
No not yet, we do `remote_client.print_and_upload_result(args.label, metrics)` in `runner.py` and then close the remote clients

Reply:
If it's the runner's responsability to reap the clients, then it should do it on error too (runner.py:152). Having the call take care of it only some of the time is surprising.

Reply:
I guess the try catch here is redundant then :)